Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectBuilder/MSBuildProjectBuilder.cs

```diff
@@ -22,14 +23,87 @@ public enum ProgressMessageType
             Error,
         }
 
-        public const string BuildTargetArgument = "/t:Build";
-        public const string RebuildTargetArgument = "/t:Rebuild";
+        public const string BuildTargetArgument = "-t:Build";
+        public const string RebuildTargetArgument = "-t:Rebuild";
         public const string DefaultBuildArguments = MSBuildProjectBuilder.BuildTargetArgument;
 
+        private static readonly Lazy<Task<string>> msBuildPathTask;
         private static readonly Regex msBuildErrorFormat = new Regex(@"^\s*(((?<ORIGIN>(((\d+>)?[a-zA-Z]?:[^:]*)|([^:]*))):)|())(?<SUBCATEGORY>(()|([^:]*? )))(?<CATEGORY>(error|warning))( \s*(?<CODE>[^: ]*))?\s*:(?<TEXT>.*)$", RegexOptions.Compiled);
 
         private static bool isBuildingWithDefaultUI = false;
 
+        static MSBuildProjectBuilder()
+        {
+            MSBuildProjectBuilder.msBuildPathTask = new Lazy<Task<string>>(async () =>
```

Comment:
>async () => [](start = 75, length = 11)

This is one very big lambda. I personally would rather see it in it's own function so that it shows up on the callstack in a more meaningful way... but I don't have a strong opposition to it remaining as a lambda.

Reply:
My preference is to make the scope as small as possible as this makes the code more self documenting. As is, there is no question at a glance that the only place this "method" is invoked is from static constructor (or rather the `Lazy` constructed there). I would be open to moving it to a local method within the static constructor, and passing that to the `Lazy` if you think that would be more readable.

Reply:
Verified call stack is readable, leaving as is.